apps/designer/app/designer/features/style-panel/style-source-input/style-source.tsx

```diff
@@ -0,0 +1,214 @@
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuPortal,
+  DropdownMenuTrigger,
+  IconButton,
+  Button,
+  Text,
+  styled,
+} from "@webstudio-is/design-system";
+import { ChevronDownIcon } from "@webstudio-is/icons";
+import {
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  type KeyboardEvent,
+  type KeyboardEventHandler,
+  type FocusEvent,
+} from "react";
+
+// Used to schedule function calls to be executed after at a later point in time.
+// Since menu is managing focus, we need to execute that callback when the management is done.
+const useCallScheduler = () => {
+  const ref = useRef<Array<() => void>>([]);
+  const call = () => {
+    for (const fn of ref.current) {
+      fn();
+    }
+    ref.current = [];
+  };
+  const add = (fn: () => void) => () => {
+    ref.current.push(fn);
+  };
+
+  return { call, add };
+};
+
+type MenuProps = {
+  onEdit: () => void;
+  onDuplicateItem: () => void;
+  onRemoveItem: () => void;
+};
+
+const Menu = (props: MenuProps) => {
+  const scheduler = useCallScheduler();
+  return (
+    <DropdownMenu modal>
+      <DropdownMenuTrigger asChild>
+        <IconButton
+          aria-label="Menu Button"
+          css={{ position: "absolute", right: 0 }}
+        >
+          <ChevronDownIcon />
+        </IconButton>
```

Comment:
Not sure as there's no menu in Figma, but I think you want `DeprecatedIconButton` in this case.

`<IconButton>` implements this component (It's intended for css styles' controls):

<img width="266" alt="Screenshot 2023-01-09 at 13 16 16" src="https://user-images.githubusercontent.com/825702/211274784-cf12687c-8318-4c4e-9156-47296c8dc944.png">

For regular button with an icon, we're going to use `Button` (but it's not ready yet, and currently `DeprecatedIconButton` is the replacement):

<img width="615" alt="Screenshot 2023-01-09 at 13 19 08" src="https://user-images.githubusercontent.com/825702/211275267-33b8468c-c068-4397-b269-3bf1419c2eda.png">

Reply:
Yeah I used it temporarily, @taylornowotny needs to finish the design there since its not quite clear what to use, but you are probably right, I just didn't want to add new usages of deprecated things. Since this is still work in progress it doesn't really matter which one we use as the design there is absolutely temporary.

Reply:
adding a comment to the current usage of IconButton